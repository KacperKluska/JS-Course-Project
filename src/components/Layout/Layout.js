import { useRef, useState } from 'react';
import { Switch, Route } from 'react-router-dom';
import Header from '../Header/Header';
import Navbar from '../Navbar/Navbar';
import Footer from '../Footer/Footer';
import Search from '../Search/Search';
import '../../fontello/css/fontello.css';
import MainPage from '../Pages/MainPage/MainPage';
import Login from '../Login/Login';
import Register from '../Register/Register';
import EmailButton from '../EmailButton/EmailButton';
import ErrorPage from '../Pages/ErrorPage/ErrorPage';
import AccountPage from '../Pages/AccountPage/AccountPage';

export default function Layout() {
  const [searchVisible, setSearchVisible] = useState('none');
  const searchRef = useRef();
  const handleSearchVisible = () => {
    setSearchVisible((prev) => {
      if (prev === 'active') {
        setTimeout(() => {
          setSearchVisible('none');
        }, 500);
        return 'disable';
      }
      return 'active';
    });
  };

  return (
    <div>
      <Header
        searchRef={searchRef}
        searchVisible={searchVisible}
        setSearchVisible={handleSearchVisible}
      />
      <Navbar />
      <Search
        searchRef={searchRef}
        styleClass={searchVisible}
        onClick={handleSearchVisible}
      />
      <main>
        <Switch>
          <Route path="/" exact>
            <MainPage />
          </Route>
          <Route path="/login">
            <Login />
          </Route>
          <Route path="/register">
            <Register />
          </Route>
          <Route path="/account">
            <AccountPage />
          </Route>
          <Route path="*">
            <ErrorPage errorCode="404 Page not found!">
              Try to go back.
            </ErrorPage>
          </Route>
        </Switch>
      </main>
      <EmailButton />
      <Footer />
    </div>
  );
}
