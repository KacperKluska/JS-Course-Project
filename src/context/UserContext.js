import React from 'react';
import { verifyUser } from '../services/requests';

export const UserContext = React.createContext('');

export const UserContextProvider = ({ children }) => {
  const [userLogged, setUserLogged] = React.useState(false);

  React.useEffect(async () => {
    try {
      const response = await verifyUser();
      if (response.status === 200) {
        setUserLogged(true);
      } else {
        setUserLogged(false);
      }
    } catch (err) {
      console.error(err);
    }
  }, [userLogged]);

  return (
    <UserContext.Provider value={{ isLogged: [userLogged, setUserLogged] }}>
      {children}
    </UserContext.Provider>
  );
};
