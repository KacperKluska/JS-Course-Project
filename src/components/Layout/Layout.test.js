import { BrowserRouter } from 'react-router-dom';
import Layout from './Layout';
import { UserContextProvider } from '../../context/UserContext';

let wrapper;

beforeAll(() => {
  wrapper = mount(
    <UserContextProvider>
      <BrowserRouter>
        <Layout />
      </BrowserRouter>
    </UserContextProvider>,
  );
});

test('layout snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});
