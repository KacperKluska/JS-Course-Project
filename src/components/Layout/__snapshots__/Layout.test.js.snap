// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout snapshot 1`] = `
<div>
  <header
    id="mainHeader"
  >
    <div
      className=""
      id="outsideAlert"
    >
      <div
        id="leftNavDiv"
      >
        <ul
          className=""
          id="leftNav"
        >
          <a
            className="link"
            href="/login"
            onClick={[Function]}
          >
            <li
              className="navItem"
            >
              <i
                className="icon-login"
              />
              Sign in
            </li>
          </a>
          <a
            className="link"
            href="/register"
            onClick={[Function]}
          >
            <li
              className="navItem"
            >
              <i />
              Register
            </li>
          </a>
          <li
            className="navItem"
            onClick={[Function]}
          >
            <i
              className="icon-search"
            />
            Search
          </li>
        </ul>
      </div>
    </div>
    <nav
      id="firstNavbar"
    >
      <div
        className="hamburgerButton"
        onClick={[Function]}
      >
        <span
          className="bar"
        />
        <span
          className="bar"
        />
        <span
          className="bar"
        />
      </div>
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        <h1
          id="logo"
        >
          SuitUp
        </h1>
      </a>
      <ul
        id="rightNav"
      >
        <li
          className="navItem"
        >
          <i
            className="icon-gift"
          />
          Gift Certificate
        </li>
        <li
          className="navItem"
        >
          <i
            className="icon-basket"
          />
          Cart
        </li>
      </ul>
    </nav>
  </header>
  <nav
    className="sticky"
  >
    <div
      className="secondNavbar"
    >
      <ul
        className="thirdList"
      >
        <li
          className="navItem"
        >
          <i />
          For him
        </li>
        <li
          className="navItem"
        >
          <i />
          For her
        </li>
        <li
          className="navItem"
        >
          <i />
          Kids
        </li>
        <li
          className="navItem"
          onClick={[Function]}
        >
          <i />
          About
        </li>
        <li
          className="navItem"
          onClick={[Function]}
        >
          <i />
          Contact
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="none"
    id="search"
  >
    <li
      className="navItem"
      onClick={[Function]}
    >
      <i
        className="icon-cancel"
      />
      Close
    </li>
    <input
      id="searchInput"
      placeholder="Search..."
      type="text"
    />
  </div>
  <main>
    <div>
      <div
        className="mainPhoto"
      >
        <img
          alt="Barney Stinson"
          src="barney5.jpg"
        />
      </div>
      <section
        className="infoText"
        id="About"
      >
        <header>
          <h1>
            About
          </h1>
        </header>
        <div
          className="text"
        >
          <p>
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Illo velit nulla eius ducimus quia suscipit minus consectetur eaque, in iusto sapiente unde voluptate alias libero delectus nesciunt animi minima totam? Obcaecati molestiae officiis vero totam quas similique aliquid deserunt, ipsa odio minima inventore cumque. Eligendi aliquid hic earum veniam dolorum doloribus sequi quidem nisi obcaecati quibusdam iusto quam quae aut quis assumenda, tempora animi illo iste dolorem aliquam? Voluptates atque iusto totam exercitationem et omnis fugit facere enim ipsa quia nisi harum voluptatem, accusantium blanditiis voluptas eum veniam amet numquam. Voluptate amet laudantium soluta. Assumenda, sapiente commodi maxime voluptatibus facere beatae at vero quae culpa provident, ea inventore omnis nemo nesciunt suscipit cum ipsum autem iure eos dolorum temporibus facilis! Fugit maiores aliquam vel explicabo non et praesentium suscipit hic eveniet repudiandae odio corrupti quisquam laboriosam dicta commodi voluptas beatae neque asperiores nam, ipsum autem corporis officia soluta? Dolore, inventore!
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque illum maxime quae qui suscipit, labore recusandae! Aperiam, iste quae consectetur, iusto repudiandae, adipisci asperiores odit libero amet sunt inventore minus? Deleniti repudiandae, odit commodi ea dolor iure fuga vitae nostrum accusamus qui doloremque molestias, totam veritatis aperiam nobis placeat soluta!
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Facilis enim, sapiente illo dignissimos soluta suscipit corporis natus nemo repellat, quam inventore sequi! Assumenda aliquam cum cupiditate ipsum blanditiis reprehenderit consectetur. Illo voluptatibus pariatur dolores aliquam explicabo, odio soluta. Blanditiis illo animi numquam non aperiam quas quasi excepturi atque laudantium impedit.
          </p>
        </div>
      </section>
      <div
        className="imagesDiv"
      >
        <h1
          className="title"
        >
          <b>
            Check out our new collection
          </b>
        </h1>
        <div
          className="gridCollage"
        >
          <div
            className="photo headerImg"
          >
            <img
              alt="Suit 1"
              src="barney6.jpg"
            />
          </div>
          <div
            className="photo asideImg"
          >
            <img
              alt="Suit 1"
              src="barney3.jpg"
            />
          </div>
          <div
            className="photo normalIMG"
          >
            <img
              alt="Suit 1"
              src="suit1.jpg"
            />
          </div>
          <div
            className="photo normalIMG"
          >
            <img
              alt="Suit 2"
              src="suit2.jpg"
            />
          </div>
          <div
            className="photo normalIMG"
          >
            <img
              alt="Suit 3"
              src="suit3.jpg"
            />
          </div>
        </div>
      </div>
      <section
        className="infoText"
        id="Contact"
      >
        <header>
          <h1>
            Contact
          </h1>
        </header>
        <div
          className="text"
        >
          <p>
            "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Libero quisquam omnis fugit hic, nobis eum sit natus ipsum voluptas recusandae commodi molestias temporibus, non voluptatem incidunt dignissimos, suscipit assumenda? Ut, assumenda maxime. Temporibus est perferendis nemo qui accusantium eum, excepturi facilis consectetur totam ex fuga libero in incidunt architecto nam.
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Itaque autem laborum nobis laudantium vitae voluptatem asperiores omnis quam mollitia dolorum?
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Quam accusamus sapiente eaque hic, laudantium consequuntur consequatur ipsa deserunt reprehenderit asperiores perferendis voluptatibus, doloribus animi cumque optio iure veniam, eveniet culpa aspernatur. Amet, nostrum similique ratione, atque, est adipisci aliquam enim sapiente quasi iure provident ea deserunt officia perspiciatis doloremque recusandae.
          </p>
        </div>
      </section>
    </div>
  </main>
  <div
    className="fixedToRight"
  >
    <button
      className="emailButton"
      onClick={[Function]}
    >
      <i
        className="icon-mail"
      />
    </button>
  </div>
  <footer
    className="footer"
  >
    <div
      className="footerCollumns"
    >
      <div
        className="footerCollumn"
      >
        <b>
          <h3>
            Help
          </h3>
        </b>
        <label>
          Returns
        </label>
        <label>
          Change
        </label>
        <label>
          FAQ
        </label>
      </div>
      <div
        className="footerCollumn"
      >
        <b>
          <h3>
            Payment and shipment
          </h3>
        </b>
        <label>
          Payment forms
        </label>
        <label>
          Shipment forms
        </label>
      </div>
      <div
        className="newsletter"
      >
        <b>
          <h3>
            Sign in for our Newsletter
          </h3>
        </b>
        <label
          style={
            Object {
              "textAlign": "start",
            }
          }
        >
          Sign in if you want to be informed about sales and new collections.
        </label>
        <form
          className="newsletterForm"
          onSubmit={[Function]}
        >
          <input
            className="newsletterInput"
            onChange={[Function]}
            placeholder="Your email..."
            required={true}
            type="email"
            value=""
          />
          <button
            className="newsletterButton"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </div>
    </div>
    <div
      className="socialMedias"
    >
      <h3>
        Join us!
      </h3>
      <div>
        <a
          href="https://www.facebook.com/"
          target="_blanc"
        >
          <i
            className="icon-facebook-official"
          />
        </a>
        <a
          href="https://www.youtube.com/"
          target="_blanc"
        >
          <i
            className="icon-youtube"
          />
        </a>
        <a
          href="https://twitter.com/?lang=pl"
          target="_blanc"
        >
          <i
            className="icon-twitter"
          />
        </a>
        <a
          href="https://www.google.com/"
          target="_blanc"
        >
          <i
            className="icon-gplus"
          />
        </a>
      </div>
    </div>
    <b>
      Copyrigth © Kacper Kluska 2021. JavaScript course
    </b>
  </footer>
</div>
`;
